opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/ScriptUtils.java

```diff
@@ -22,7 +21,8 @@ public class ScriptUtils {
    * Limitation: assume inner field name is always "keyword".
    */
   public static String convertTextToKeyword(String fieldName, ExprType fieldType) {
-    if (fieldType == OPENSEARCH_TEXT_KEYWORD) {
+    if (fieldType instanceof OpenSearchTextType
+        && ((OpenSearchTextType) fieldType).getFields().size() > 0) {
```

Comment:
No, it is not safe. In case if user makes search on text field without `fields`/`fielddata`, OpenSearch engine raises an exception like
> Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [<field>] in order to load field data by uninverting the inverted index.